go/p2p/discovery/bootstrap/client.go

```diff
@@ -133,6 +133,13 @@ func (c *client) Advertise(ctx context.Context, ns string, _ ...discovery.Option
 
 	pf.RecordSuccess()
 
+	// Close connections after every call because requests to the seed node are infrequent.
```

Comment:
We should also do the same on the server side to handle clients that don't close connections.

Reply:
Agree, but its a bit uglier because currently the server only handles streams. And I'm not sure if the caller will receive the data if the connection is closed immediately after the stream.

Reply:
@kostko  Tested locally. If connection is closed after the stream, the caller may receive an error indicating that the stream was reset. Therefore, if the server intends to close connections, it should wait a few seconds before doing so.

Connection handling is the responsibility of the connection manager. To speed up connection closing, we can set the `PeerGracePeriod` configuration for seed nodes to 2 seconds (default 20 seconds), for instance. Optionally, we might consider introducing default configurations for various roles.

Reply:
> Connection handling is the responsibility of the connection manager. To speed up connection closing, we can set the PeerGracePeriod configuration for seed nodes to 2 seconds (default 20 seconds), for instance.

Sounds like the cleanest solution to me.

Reply:
No, connection is not closed after 2 secs. It is just out of the grace period and if the number of connections is 130+ (equal or above high watermark) it can be closed.

If we want to keep the number of connections low, we would need to set low watermark to 1 and high to 2. This way, all connections but 1 will be closed after 2 seconds. Currently, we cannot set this in the config as we hard coded difference between low and high (30), so changes to the config are needed.

Reply:
Ok, that's good as well (e.g. that connections are only closed under stress, as long as this results in new connections succeeding).